x/payments/spec/05_events.md

```diff
@@ -0,0 +1,60 @@
+
+## Handler
+
+## MsgCreatePaymentTemplate
+| Type                            | Attribute Key            | Attribute Value       |
+|---------------------------------|--------------------------|-----------------------|
+| create_payment_template         | creator_did              | {creator_did}         |
+| create_payment_template         | id                       | {id}                  |
+| create_payment_template         | payment_amount           | {payment_amount}      |
+| create_payment_template         | payment_minimum          | {payment_minimum}     |
+| create_payment_template         | payment_maximum,         | {payment_maximum}     |
+| create_payment_template         | discounts                | {discounts}           |
+| create_payment_template         | wallet_distribution      | {wallet_distribution} |
+
+## MsgCreatePaymentContract
+| Type                            | Attribute Key            | Attribute Value       |
+|---------------------------------|--------------------------|-----------------------|
+| create_payment_contract         | creator_did              | {creator_did}         |
+| create_payment_contract         | payment_template_id      | {payment_template_id} |
+| create_payment_contract         | payment_contract_id      | {payment_contract_id} |
+| create_payment_contract         | payer                    | {payer}               |
+| create_payment_contract         | discount_id              | {wallet_distribution} |
+| create_payment_contract         | can_deauthorise          | {can_deauthorise}     |
+
+## MsgGrantDiscount
+| Type                            | Attribute Key            | Attribute Value       |
+|---------------------------------|--------------------------|-----------------------|
+| grant_discount                  | sender_did               | {sender_did}          |
+| grant_discount                  | payment_contract_id      | {payment_contract_id} |
+| grant_discount                  | discount_id              | {discount_id}         |
+| grant_discount                  | Recipient                | {Recipient}           |
+
+## MsgRevokeDiscount
+| Type                            | Attribute Key            | Attribute Value       |
+|---------------------------------|--------------------------|-----------------------|
+| revoke_discount                 | sender_did               | {sender_did}          |
+| revoke_discount                 | payment_contract_id      | {payment_contract_id} |
+| revoke_discount                 | holder                   | {holder}              |
+
+## MsgEffectPayment
+| Type                            | Attribute Key            | Attribute Value       |
+|---------------------------------|--------------------------|-----------------------|
+| effect_payment                  | sender_did               | {sender_did}          |
+| effect_payment                  | payment_contract_id      | {payment_contract_id} |
```

Comment:
Please copy these under EndBlocker as well, as both EndBlocker and MsgEffectPayment generate these.

Reply:
effectpayment